tests_integration/lib/integration_test_helper.py

```diff
@@ -0,0 +1,166 @@
+
+        # Initialize a new git repository and commit everything
+        subprocess.run(
+            ["git", "init", f"--initial-branch={branch}"], cwd=dest_dir, check=True
+        )
+        subprocess.run(["git", "add", "--all"], cwd=dest_dir, check=True)
+        subprocess.run(
+            ["git", "commit", "-m", "Overwrite Repository Contents"],
+            cwd=dest_dir,
+            check=True,
+        )
+
+        # Push the new contents to the repository
+        push_token = get_github_token(owner_account)
+        run_command_hiding_token(
+            [
+                "git",
+                "push",
+                f"https://{push_token}@github.com/{owner_account}/{repo_name}.git",
+                branch,
+                "--force",
+            ],
+            token=push_token,
+            cwd=dest_dir,
+            check=True,
+        )
+
+        LOGGER.info(
+            f"Repository contents of {repo_name} have been overwritten successfully."
+        )
+
+    @staticmethod
+    def assert_version_pr_present(
+        feedstock: str, new_version: str, new_hash: str, old_version: str, old_hash: str
+    ):
+        """
+        Asserts that the bot has opened a version update PR.
+
+        :param feedstock: The feedstock we expect the PR for, without the -feedstock suffix.
+        :param new_version: The new version that is expected.
+        :param new_hash: The new SHA-256 source artifact hash.
+        :param old_version: The old version of the feedstock, to check that it no longer appears in the recipe.
+        :param old_hash: The old SHA-256 source artifact hash, to check that it no longer appears in the recipe.
+
+        :raises AssertionError: if the assertion fails
+        """
+        gh = Github(get_github_token(GitHubAccount.CONDA_FORGE_ORG))
+
+        full_feedstock_name = feedstock + FEEDSTOCK_SUFFIX
+        repo = gh.get_organization(GitHubAccount.CONDA_FORGE_ORG).get_repo(
+            full_feedstock_name
+        )
+        matching_prs = [
+            pr for pr in repo.get_pulls(state="open") if f"v{new_version}" in pr.title
+        ]
+
+        assert (
+            len(matching_prs) == 1
+        ), f"Found {len(matching_prs)} matching version PRs, but exactly 1 must be present."
+
+        matching_pr = matching_prs[0]
+
+        assert matching_pr.head.repo.owner.login == GitHubAccount.BOT_USER
+        assert matching_pr.head.repo.name == full_feedstock_name
+
+        cli = GitCli()
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            target_dir = Path(tmpdir) / full_feedstock_name
+            cli.clone_repo(matching_pr.head.repo.clone_url, target_dir)
+            cli.checkout_branch(target_dir, matching_pr.head.ref)
+
+            with open(target_dir / "recipe" / "meta.yaml") as f:
+                meta = f.read()
+
+        assert f'{{% set version = "{new_version}" %}}' in meta
+        assert f"sha256: {new_hash}" in meta
```

Comment:
should we also account for recipe.yaml here?
we could for example use https://github.com/conda-forge/rattler-build-feedstock in the tests (in the future?)

Reply:
Would it be okay for you to do that in a follow-up PR? I purposefully tried to keep this PR minimal with only one test feedstock. In a follow-up, I will add a test for a recipe.yaml feedstock. rattler-build-feedstock sounds like a good example!
I think it makes sense to wait with changes to the integration test helpers until then, to avoid introducing "dead" test code.

Reply:
yes, sounds good to me!